### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by mutinhiri**
In this PR i used Transactions to safely update and delete entries in the animals database. Rollback and commit are used to create a restore point
- In this Pr i also queried data using aggregate functions to test the implementation an use the ORDER BY and GROUP BY to manage data effectively

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,4 +8,44 @@ select date_of_birth from animals where name = 'Augmon' or name = 'Pikachu'
...
+
+select * from animals
+
+begin;
+delete from animals where date_of_birth > '2022-01-01';
+savepoint sp1
+update animals set weight_kg = weight_kg * -1
+rollback to sp1
+update animals set weight_kg = weight_kg * -1 where weight_kg < 0;
+commit
```
Comment (rukundoeric) [inline queries.sql]: - [ ] <b>REQUIRED:</b> Kindly make sure to verify that changes have been made before and after `COMMIT`. Something like this: 

```sql
   BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit

File: schema.sql [hunk 2]
```diff
@@ -7,6 +7,7 @@ CREATE TABLE animals (
     date_of_birth date,
     escape_attempts int,
     neutered boolean,
-    weight_kg real
+    weight_kg real,
+    species varchar
```
Comment (rukundoeric) [inline schema.sql]: - [ ] <b>REQUIRED:</b> You are expected to add a new column called species with a string type by using `ALTER TABLE` command, not just putting in the previous command. Kindly fix that.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @mutinhiri, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by descholar-ceo — APPROVED** [review]
Hi @mutinhiri , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**